### PR TITLE
feat: setup email templates

### PR DESCRIPTION
**Pull request by G3root**
fixes: #25

todo

- [x] invite template

### CODE REVIEW

File: src/emails/MemberInviteEmail.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+              >
+                {invitedByEmail}
+              </Link>
+              ) has invited you to the <strong>{teamName}</strong> team on{" "}
+              <strong>OpenCap</strong>.
+            </Text>
+
+            <Section className="mb-[32px] mt-[32px] text-center">
+              <Button
+                className="rounded bg-[#000000] px-5 py-3 text-center text-[12px] font-semibold text-white no-underline"
```
Comment (tedspare) [inline src/emails/MemberInviteEmail.tsx:61]: `bg-[#000000]` → `bg-black` if possible

File: src/emails/MemberInviteEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                Join the team
+              </Button>
+            </Section>
+            <Text className="!text-[14px] leading-[24px] text-black">
+              or copy and paste this URL into your browser:{" "}
+              <Link href={inviteLink} className="text-blue-600 no-underline">
+                {inviteLink}
+              </Link>
+            </Text>
+            <Hr className="mx-0 my-[26px] w-full border border-solid border-[#eaeaea]" />
```
Comment (tedspare) [inline src/emails/MemberInviteEmail.tsx:73]: `border-[#eaeaea]` → `border-neutral-200` if possible

File: src/server/auth.ts [hunk 3]
```diff
@@ -48,8 +52,23 @@ export const authOptions: NextAuthOptions = {
...
+          MagicLinkEmail({
+            magicLink: url,
+          }),
+        );
+
+        await sendMail({
+          to: identifier,
+          subject: "Your OpenCap Login Link",
+          html,
+        });
```
Comment (tedspare) [inline src/server/auth.ts:70]: Super clean, nice work

File: src/emails/MagicLinkEmail.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          </Button>
+        </Section>
+        <Text style={paragraph}>
+          or copy and paste this URL into your browser:{" "}
+          <Link href={magicLink} style={reportLink}>
+            {magicLink}
+          </Link>
+        </Text>
+        <Hr style={hr} />
+        <Link href="https://opencap.co/" style={reportLink}>
```
Comment (tedspare) [inline src/emails/MagicLinkEmail.tsx:39]: `"https://opencap.co/"` → `constants.url`

and

`import { constants } from "@/lib/constants";`

File: src/emails/MagicLinkEmail.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          OpenCap
+        </Link>
+      </Container>
+    </Body>
+  </Html>
+);
+
+MagicLinkEmail.PreviewProps = {
+  magicLink: "https://opencap.co/",
```
Comment (tedspare) [inline src/emails/MagicLinkEmail.tsx:48]: Same here, `constants.url`

**Review by tedspare — APPROVED** [review]
LGTM, awesome work Nafees 🚢

Only nitpick: could you match the login button color to the others?

Tested locally with `bun run email:preview` and it looks great. The JSX email preview feature is nice too:

image

image

File: src/emails/MagicLinkEmail.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+} as MagicLinkEmailProps;
+
+export default MagicLinkEmail;
+
+const main = {
```
Comment (dahal) [inline src/emails/MagicLinkEmail.tsx:53]: Can we use tailwind here as well, pretty please 🙏🏼

**Review by dahal — CHANGES_REQUESTED** [review]
Can we use tailwind on [MagicLinkEmail.tsx](https://github.com/opencapco/opencap.co/pull/26/files#diff-694507ba4413b6d7878fd7a918b3c7449ee85dce33b5caf68f6979a6467340e2) as well 🙏🏼

**Comment by G3root** [comment]
pushed the nits and requested changes

Screenshot 2024-01-14 at 07-24-26 JSX email • Member Invite Email

**Comment by dahal** [comment]
> pushed the nits and requested changes
> 
> Screenshot 2024-01-14 at 07-24-26 JSX email • Member Invite Email

Lets have same button as invite email (black). Its good to after that. Impressive work, as always @G3root 👏🏼

**Comment by G3root** [comment]
@dahal  Done. I read the comment the other way around. 😸 

Screenshot 2024-01-14 at 08-03-12 JSX email • Magic Link Email

**Pull request merged**